### ISSUE TITLE
Reproducibility

### ISSUE THREAD

**Issue by bvelitchkine** [valentingol/GANJax#11]
https://github.com/valentingol/GANJax/blob/f66a8707d2b2fa0298fa0257a91eeb1e4464b747/utils/data.py#L70

# Reproduire le bug

1. J'ai créé un mini dataset dans `data/CelebA/miniSet` contenant 50 images du dataset complet.
2. J'ai modifié load_images_celeba_xx dans `utils/data.py` pour loader automatiquement les images de ce mini data set.
3. J'ai effectué trois expériences :
- J'ai run 2 fois `app/train.py` avec *batch_size = 5*, *num_epoch = 1* en faisant en sorte d'enregistrer l'historique des loss du generateur et du discriminateur dans des fichiers .npy différents à chaque fois. Après les deux runs, je me retrouve avec 4 fichiers : `gen_history.npy`, `disc_history.npy`, `gen_history2.npy` et `disc_history2.npy`.
- Pour le dernier run, je suis retourné dans les data loaders que j'ai modifiés ainsi de sorte à avoir une buffer size inférieure à la taille du dataset (désormais de 50) : 
```python
dataset = dataset.shuffle(buffer_size=5, seed=seed) # Changed buffer size from 1000 to 5
```
Les historiques des loss pour cette dernières expériences ont été stockés dans : `gen_history3.npy` et `disc_history3.npy`.

# Constater les différences

Avec ce bout de code, j'ai loadé tous les historiques : 

```python
import jax.numpy as jnp

path = '' # Path to where the arrays were stored

gen_loss_history = jnp.load(
    f'{path}/gen_history.npy')
disc_loss_history = jnp.load(
   f'{path}/disc_history.npy')

gen_loss_history2 = jnp.load(
   f'{path}/gen_history2.npy')
disc_loss_history2 = jnp.load(
    f'{path}/disc_history2.npy')

gen_loss_history3 = jnp.load(
    f'{path}/gen_history3.npy')
disc_loss_history3 = jnp.load(
   f'{path}/disc_history3.npy')

print(gen_loss_history == gen_loss_history2) # Retourne du true partout
print(gen_loss_history == gen_loss_history3) # Retourne presque que du False
```

# Conclusion

Je pense donc que tu avais raison @valentingol, le pb venait du chargement des images avec tf et de cette fameuse buffer size qui doit être de taille supérieure à celle du dataset d'après [la doc officielle](https://www.tensorflow.org/api_docs/python/tf/data/Dataset#shuffle).

**Comment by valentingol** [comment]
Changing the buffer size doesn't change the randomness behavior in my tests. Lack of reproducibility seems to be due to CuDNN (same as #4823, #565). I'll reopen it later if a solution appears.
